shenyu-plugin/shenyu-plugin-sign/src/main/java/org/apache/shenyu/plugin/sign/service/DefaultSignService.java

```diff
@@ -123,16 +125,11 @@ private Pair<Boolean, String> sign(final ShenyuContext shenyuContext, final Serv
         return Pair.of(Boolean.TRUE, "");
     }
 
-    private Map<String, String> buildParamsMap(final ShenyuContext shenyuContext, final Map<String, Object> requestBody) {
-        Map<String, String> map = Maps.newHashMapWithExpectedSize(3);
-        map.put(Constants.TIMESTAMP, shenyuContext.getTimestamp());
-        map.put(Constants.PATH, shenyuContext.getPath());
-        map.put(Constants.VERSION, "1.0.0");
-        if (!ObjectUtils.isEmpty(requestBody)) {
-            requestBody.forEach((key, value) -> {
-                map.putIfAbsent(key, Objects.toString(value, null));
-            });
-        }
-        return map;
+    private String buildExtSignKey(final String signKey, final ShenyuContext shenyuContext) {
+        return String.join("", Constants.TIMESTAMP, shenyuContext.getTimestamp(), Constants.PATH, shenyuContext.getPath(), Constants.VERSION, "1.0.0", signKey);
+    }
+
+    private Map<String, String> transStringMap(final Map<String, Object> map) {
+        return map == null ? null : map.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> Objects.toString(e.getValue(), null)));
```

Comment:
you need refactor this code

Reply:
Tanks, can you please tell me what's wrong with this code?I need your advice. I don't understand how to refactor

Reply:
please refer https://shenyu.apache.org/zh/community/code-suggestions

Reply:
How about deleting this private method. 

```java
        final Map<String,String> requestBodyStrMap =  requestBody == null ? null : requestBody.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> Objects.toString(e.getValue(), null)));
        final String extSignKey = String.join("", Constants.TIMESTAMP, shenyuContext.getTimestamp(), Constants.PATH, shenyuContext.getPath(), Constants.VERSION, "1.0.0", appAuthData.getAppSecret());
        final String sign = ShenyuSignProviderWrap.generateSign(extSignKey, requestBodyStrMap, queryParams);
        boolean result = Objects.equals(sign, shenyuContext.getSign());
```